lib/clients/metrics.ts

```diff
@@ -1,11 +1,29 @@
+import { Counter, ValueRecorder } from '@opentelemetry/api-metrics';
+import { PrometheusExporter } from '@opentelemetry/exporter-prometheus';
+import { Meter, MeterProvider } from '@opentelemetry/sdk-metrics-base';
 import { BufferedMetricsLogger } from 'datadog-metrics';
 import Hashids from 'hashids';
 
+import log from '@/logger';
 import { Config } from '@/types';
 
 export class Metrics {
```

Comment:
The `Metrics` class is missing a `shutdown` function to stop the prometheus server. We should add that function and call `metrics.shutdown()` during the shutdown process of the `serviceManager` to properly clean up all resources.

Reply:
fixed!